app/models/ext_management_system.rb

```diff
@@ -260,6 +264,78 @@ def default_endpoint
     default || endpoints.build(:role => "default")
   end
 
+  # TODO: takes multiple connection data
+  # endpoints, authentications, and works out what's what.
+  def connections=(options)
+    options.each do |option|
+      endpoint = endpoint_builder(option[:endpoint])
+      authentication_builder(option[:authentication])
+        # connection_by_role=(option)
+    end
+  end
+
+  def connections
+    roles = endpoints.map(&:role)
+    options = {}
+
+    roles.each do |role|
+      conn = connection_by_role(role)
+      options[role] = conn
+    end
+
+    connections = OpenStruct.new(options)
+    connections.roles = roles
+    connections
+  end
+
+  # Takes a hash of connection data
+  # hostname, port, and authentication
+  # if no role is passed in assume is default role
+  def connection_by_role=(options)
+    unless options[:endpoint].key?(:role)
+      options[:endpoint][:role] = "default"
+    end
+
+    endpoint_builder(options[:endpoint])
+    authentication_builder(options[:authentication])
+  end
+
+  def endpoint_builder(options)
```

Comment:
This method name, since it's a noun, makes me thing it's returning a builder object.  So, when it's read from the caller perspective, it's not clear that it's updating anything, or what it even does.  Perhaps `build_endpoint`?  Or, if Rails eats that method, maybe something like `build_endpoint_by_role` or `update_endpoint` or `ensure_endpoint` or `ensure_endpoint_by_role` (you get the idea :wink:)

Also, should it be a private method?  I'm not sure anything outside this class would use it directly.